evmcore/tx_validation.go

```diff
@@ -125,19 +116,73 @@ func validateTx(tx *types.Transaction, opt validationOptions) error {
 		return ErrTipAboveFeeCap
 	}
 
-	// Make sure the transaction is signed properly.
-	from, err := types.Sender(opt.signer, tx)
-	if err != nil {
-		return ErrInvalidSender
+	// Sonic only supports Blob transactions without blob data.
+	if tx.Type() == types.BlobTxType &&
+		(len(tx.BlobHashes()) > 0 || (tx.BlobTxSidecar() != nil && len(tx.BlobTxSidecar().BlobHashes()) > 0)) {
+		return ErrNonEmptyBlobTx
```

Comment:
This is not an immutable property defined by the specification of transactions and may actually change, depending on the context. 

For instance, if we would, one day, support blobs, this becomes a check to be moved to the `ValidateTxForNetwork` function. If at that point the static checks are part of the event validation steps, we need to add a network rule parameter to this function to manage the transition.